src/TabComplete.js

```diff
@@ -45,21 +48,32 @@ class TabComplete {
         this.isFirstWord = false; // true if you tab-complete on the first word
         this.enterTabCompleteTimerId = null;
         this.inPassiveMode = false;
+        this.memberTabOrder = {};
+        this.memberOrderSeq = 0;
     }
 
     /**
-     * @param {Entry[]} completeList
+     * Call this when a a UI element representing a tab complete entry has been clicked
+     * @param {entry} The entry that was clicked
      */
-    setCompletionList(completeList) {
-        this.list = completeList;
+    onEntryClick(entry) {
         if (this.opts.onClickCompletes) {
-            // assign onClick listeners for each entry to complete the text
-            this.list.forEach((l) => {
-                l.onClick = () => {
-                    this.completeTo(l);
-                }
-            });
+            this.completeTo(entry);
+        }
+    }
+
+    loadEntries(room) {
```

Comment:
Yes, noticed this was designed to be independent from what it was completing. It would have required a lot more fiddling to maintain this property whilst not re-doing the search every time. This means we can just pass around the TabComplete object rather than sometimes the tabcomplete object and sometimes the entries. If we ever actually want to use tabcomplete elsewhere, we can rearrange again.